backends/cuda-gen/ceed-cuda-gen-operator.c

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2017-2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory. LLNL-CODE-734707.
+// All Rights reserved. See files LICENSE and NOTICE for details.
+//
+// This file is part of CEED, a collection of benchmarks, miniapps, software
+// libraries and APIs for efficient high-order finite element and spectral
+// element discretizations for exascale applications. For more information and
+// source code availability see http://github.com/ceed.
+//
+// The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+// a collaborative effort of two U.S. Department of Energy organizations (Office
+// of Science and the National Nuclear Security Administration) responsible for
+// the planning and preparation of a capable exascale ecosystem, including
+// software, applications, hardware, advanced system engineering and early
+// testbed platforms, in support of the nation's exascale computing imperative.
+
+#include <ceed-backend.h>
+#include "ceed-cuda-gen.h"
+#include "ceed-cuda-gen-operator-build.h"
+#include "../cuda/ceed-cuda.h"
+
+static int CeedOperatorDestroy_Cuda_gen(CeedOperator op) {
+  int ierr;
+  CeedOperator_Cuda_gen *impl;
+  ierr = CeedOperatorGetData(op, (void *)&impl); CeedChk(ierr);
+
+  // for (CeedInt i = 0; i < impl->numein + impl->numeout; i++) {
+  //   ierr = CeedVectorDestroy(&impl->evecs[i]); CeedChk(ierr);
+  // }
+  // ierr = CeedFree(&impl->evecs); CeedChk(ierr);
+  // ierr = CeedFree(&impl->edata); CeedChk(ierr);
+
+  // for (CeedInt i = 0; i < impl->numein; i++) {
+  //   ierr = CeedVectorDestroy(&impl->qvecsin[i]); CeedChk(ierr);
+  // }
+  // ierr = CeedFree(&impl->qvecsin); CeedChk(ierr);
+
+  // for (CeedInt i = 0; i < impl->numeout; i++) {
+  //   ierr = CeedVectorDestroy(&impl->qvecsout[i]); CeedChk(ierr);
+  // }
+  // ierr = CeedFree(&impl->qvecsout); CeedChk(ierr);
+
+  // ierr = CeedFree(&impl->W); CeedChk(ierr);
```

Comment:
Should this commented code be executed?

Reply:
Thanks, I completely forgot about the destructors.